support/cas-server-support-discovery-profile/src/main/java/org/apereo/cas/discovery/CasServerProfileRegistrar.java

```diff
@@ -122,6 +124,13 @@ private Object locateSubtypesByReflection(final Function<Class, Object> mapper,
         return clients.findAllClients().stream().map(Client::getName).collect(Collectors.toSet());
     }
 
+    private Set<String> locateAvailableAttributes() {
+        if (attributeRepository == null) {
+            return new LinkedHashSet<>(0);
+        }
+        return attributeRepository.getPossibleUserAttributeNames();
```

Comment:
It might also be good to combine this with attributes defined to be fetched directly from authn sources. (LDAP and JDBC at this point).